documentation/book/assembly-upgrade.adoc

```diff
@@ -3,20 +3,83 @@
 // master.adoc
 
 [id='assembly-upgrade-{context}']
-= {ProductName} and Kafka upgrades
+= Upgrade
 
-Each version of {ProductName} supports a range of versions of Apache Kafka.
-You can upgrade to a higher Kafka version as long as that version is supported by your version of {ProductName}.
+You can upgrade {ProductName} with no cluster downtime. Each version of {ProductName} supports a range of versions of Apache Kafka. You can upgrade to a higher Kafka version as long as that version is supported by your version of {ProductName}.
 In some cases, you can also downgrade to a lower supported Kafka version.
 
-When a newer version of {ProductName} is available, it may provide support for newer versions of Kafka. 
-Therefore, you will need to upgrade to the new version of {ProductName} before you can upgrade to a higher supported Kafka version.
-Upgrading the version of {ProductName} is done by upgrading the Cluster Operator deployment to the new version.
+Later versions of {ProductName} provide support for newer versions of Kafka, so you need to upgrade {ProductName} _before_ you can upgrade to a higher supported Kafka version.
 
 ifdef::StrimziUpgrades[]
-include::proc-upgrading-the-cluster-operator-0-10-0-to-0-11-0.adoc[leveloffset=+1]
+*Kafka versions*
+
+The Kafka message format version and inter-broker protocol version are subject to change. As a result, the upgrade process involves making both configuration changes to existing Kafka brokers and code changes to client applications (consumers and producers). The following table shows the differences between the Kafka versions:
+
+include::snip-kafka-versions.adoc[leveloffset=+1]
+
+endif::StrimziUpgrades[]
+
+ifndef::StrimziUpgrades[]
```

Comment:
This is one of the sections which shouldn't really be here because they belong downstream. Should we start doing something about it already?

Reply:
We are going to plan in the work as separate issues for RHEL and OCP. One thing that could really help is just showing the oc command in the upstream procedures. Do we really have to duplicate all the commands? We could have an upstream snippet at the start of the doc that states that you can use either oc or kubectl, but for 'clarity' we're just showing oc in the examples?

Reply:
I see two problems with it:
* I'm not sure the commands are always the same
* If we would need to decide for one, it would probably be `kubectl` and not `oc` which might make your life easier.

(BTW: This comment is for me not a blocker for merging this PR - I just wanted to point out that we will need to address it in the future. But I'm fine with merging this.)

Reply:
Thanks Jakub. We could have kubectl upstream and oc downstream using a variable. We can give it some thought.

Reply:
I still don't understand why we're sticking with the strategy of documenting both platforms in the same book. No one has given a good reason why we shouldn't just produce two books, one for OCP and one for Kubernetes. Many of the examples could just be dealt with via an attribute as @PaulRMellor suggests. It would make the docs easier to maintain and easier for the user to understand.

Reply:
I am a HUGE fan of not having 1 book for everything! I think it was a request early on, but is causing all kinds of issues.  @jenmalloy can talk to having 2 (or more) books from her experience with EnMasse.

Reply:
I have created a separate issue to track the conversation on the upstream/downstream split:

https://issues.jboss.org/browse/ENTMQST-919

Please add any thoughts there. We can use this PR to get the upgrade content in and treat the re-architecture separately.